File: simpleFlowSolver2D/calcFluxConCDS.jl
function calcFluxConCDS!(fluxConX, fluxConY, Phi, U, V, Dx)

	# using central differencing
	
	nG = 1

	ImaAll, JmaAll = size(fluxConX)
	Ifi = nG+1; Ifim = Ifi-1; Ifip = Ifi+1
	Ila = ImaAll -nG; Ilap = Ila+1; Ilam = Ila-1
	Jfi = nG+1; Jfim = Jfi-1; Jfip = Jfi+1
	Jla = JmaAll -nG; Jlap = Jla+1; Jlam = Jla-1

	PhiC = @view Phi[Ifi:Ila, Jfi:Jla]
	PhiE = @view Phi[Ifip:Ilap, Jfi:Jla]; PhiW = @view Phi[Ifim:Ilam, Jfi:Jla]
	PhiN = @view Phi[Ifi:Ila, Jfip:Jlap]; PhiS = @view Phi[Ifi:Ila, Jfim:Jlam]

	Ue = @view U[Ifi:Ila, Jfi:Jla]; Uw = @view U[Ifim:Ilam, Jfi:Jla]
	Vn = @view V[Ifi:Ila, Jfi:Jla]; Vs = @view V[Ifi:Ila, Jfim:Jlam]

	fluxConX[Ifi:Ila, Jfi:Jla] .= 0.5 *Dx^2 *((PhiE +PhiC) .*Ue .- (PhiC +PhiW) .*Uw)

	fluxConY[Ifi:Ila, Jfi:Jla] .= 0.5 *Dx^2 *((PhiN +PhiC) .*Vn .- (PhiC + PhiS) .*Vs)
	
	return nothing
end

File: nfs2024_matlab/simpleFlowSolver2D/calcFluxConCDS.m
% # net.m ###############################################
% Programm: 		    	simpleFlowSolver2D
% Content:					convective fluxes clac CDS 
% Author:                   S.H. Tey, June 2024
% Email:                    sheu.tey@stud.uni-due.de
% Matriculation Number:     3117568
% -------------------------------------------------------
% function for convective fluxes calculation with CDS
% differencing scheme:
%   - Phi(:,:) is stored in face centre; 
%   - U(:,:) and V(:,:) are stored in face surfaces
% =======================================================

function [fluxConX, fluxConY] = calcFluxConCDS( ... 
    fluxConX,fluxConY, Phi, U, V) % #####################
% ------- parameters ------------------------------------
global Ifi Ifim Ifip Ila Ilam Ilap
global Jfi Jfim Jfip Jla Jlam Jlap
global Delta
% ------- preallocating and preindexing -----------------
PhiC = Phi(Ifi:Ila, Jfi:Jla);
PhiE = Phi(Ifip:Ilap, Jfi:Jla);
PhiW = Phi(Ifim:Ilam, Jfi:Jla);
PhiN = Phi(Ifi:Ila, Jfip:Jlap);
PhiS = Phi(Ifi:Ila, Jfim:Jlam);

Ue = U(Ifi:Ila, Jfi:Jla);
Uw = U(Ifim:Ilam, Jfi:Jla);
Vn = V(Ifi:Ila, Jfi:Jla);
Vs = V(Ifi:Ila, Jfim:Jlam);
% ------- operation -------------------------------------
fluxConX(Ifi:Ila, Jfi:Jla) = 0.5 *Delta * ( ...
    (PhiE +PhiC).*Ue - (PhiC +PhiW).*Uw );
fluxConY(Ifi:Ila, Jfi:Jla) = 0.5 *Delta * ( ...
    (PhiN +PhiC).*Vn - (PhiC +PhiS).*Vs );
end % ###################################################
